### PR TITLE
Bluetooth: ascs: Drop ISO PDUs if ASE is not in streaming state

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This fixes missing drop of ISO Data PDUs received in non-streaming state.
The server shall indicate first it's readiness to receive the ISO Data
by calling bt_bap_stream_start that triggers state transition from
Enabling to Streaming state.

This adds tests for receiving ISO Data PDU.
* test_recv_in_streaming_state - test whether received ISO Data PDU is
  sent to upper layers.
* test_recv_in_enabling_state - test whether received ISO Data PDu is
  dropped because upper layer is not ready to receive the data yet.

### CODE REVIEW

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -671,10 +671,13 @@ static void ascs_iso_recv(struct bt_iso_chan *chan,
...
-	    ep->status.state != BT_BAP_EP_STATE_STREAMING) {
-		LOG_DBG("ep %p is not in the streaming state: %s", ep,
-			bt_bap_ep_state_str(ep->status.state));
+	if (ep->status.state != BT_BAP_EP_STATE_STREAMING) {
+		if (IS_ENABLED(CONFIG_BT_BAP_DEBUG_STREAM_DATA)) {
+			LOG_DBG("ep %p is not in the streaming state: %s", ep,
+				bt_bap_ep_state_str(ep->status.state));
+		}
+
+		net_buf_unref(buf);
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:680]: Is the `net_buf_unref` correct to use here? IIRC it is no, and if it is, it needs to be added to _all_ cases.

File: subsys/bluetooth/audio/ascs.c [hunk 2]
```diff
@@ -671,10 +671,13 @@ static void ascs_iso_recv(struct bt_iso_chan *chan,
 		return;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_BAP_DEBUG_STREAM_DATA) &&
-	    ep->status.state != BT_BAP_EP_STATE_STREAMING) {
-		LOG_DBG("ep %p is not in the streaming state: %s", ep,
-			bt_bap_ep_state_str(ep->status.state));
+	if (ep->status.state != BT_BAP_EP_STATE_STREAMING) {
+		if (IS_ENABLED(CONFIG_BT_BAP_DEBUG_STREAM_DATA)) {
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:675]: The issue you are fixing here by splitting the `if` is also missing in the unicast client. Could you update that too please?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/ascs.c (hunk 2 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c:675]: sure

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c:680]: That's surprise :) I see that there is no unref in samples as well. I thought that stack passes the buffer ownership

Comment (Thalley) [reply subsys/bluetooth/audio/ascs.c]: The stack unrefs the buf in `bt_iso_recv` by calling `bt_conn_reset_rx_state(iso);`

**Review by Thalley — APPROVED** [review]

**Review by larsgk — APPROVED** [review]

**Pull request merged**